File: 김가은/WEEK1.js
// 최종 풀이 1
function solution(s) {
  let answer = 0;
  let arr = ["()", "[]", "{}"];

  for (let i = 0; i < s.length; i++) {
    let stack = [];
    for (let x of s) {
      arr.includes(stack.at(-1) + x) ? stack.pop() : stack.push(x);
    }
    if (!stack.length) answer++;
    s = (s + s[0]).slice(1);
  }
  return answer;
}

// 최종 풀이 2
function solution2(s) {
  let answer = 0;

  for (let i = 0; i < s.length; i++) {
    let tempS = s;

    while (/(\(\)|{}|\[\])/g.test(tempS)) {
      tempS = tempS.replace(/(\(\)|{}|\[\])/g, "");
    }
    if (tempS.length === 0) answer++;
    s = (s + s[0]).slice(1);
  }
  return answer;
}

File: 이제준/src/WEEK1.java
import java.util.*;

class Solution {
    public int solution(String s) {
        int answer = 0;
        String str = s;

        for (int i = 0; i < s.length(); i++) {
            if (checkBracket(str)) {
                answer++;
            }
            str = str.substring(1, s.length()) + str.charAt(0);
        }
        return answer;
    }

    private boolean checkBracket(String str) {
        Stack<Character> s = new Stack<>();
        char[] arr = str.toCharArray();

        for (char c : arr) {
            if (s.isEmpty()) {
                s.push(c);
            } else {
                if (s.peek() == '[' && c == ']' || s.peek() == '(' && c == ')' || s.peek() == '{' && c == '}') {
                    s.pop();
                } else {
                    s.push(c);
                }
            }
        }
        return s.isEmpty();
    }
}